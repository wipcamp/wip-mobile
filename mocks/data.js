export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]