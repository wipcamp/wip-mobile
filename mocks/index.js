const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }