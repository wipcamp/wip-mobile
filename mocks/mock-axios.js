import axios from 'axios'
import mockAxios from 'axios-mock-adapter'

import env from '../config'

const mock = new mockAxios(axios)

export default mock