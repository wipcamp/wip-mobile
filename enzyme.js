import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { initAll } from './mocks'

configure({ adapter: new Adapter() })
initAll()