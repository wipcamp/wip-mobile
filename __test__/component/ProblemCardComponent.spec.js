/**
 * @jest-environment node
 */

import React from 'react'
import { shallow } from 'enzyme'

import { problem } from '../../mocks/data'
import ProblemCard from '../../components/ProblemCardComponent'
// import styles from '../../styles/ViewProblemStyle'
// import reportStyle from '../../styles/reportProblemStyle'

describe('rendering', () => {
    describe('container', () => {
        let props
        let wrapper
        beforeEach(() => {
            props = {
                data: problem[0],
                navigation: jest.fn()
            }
            wrapper = shallow(<ProblemCard {...props} />)
        })
        it('should have TouchableOpacity', () => {
            expect(wrapper.find('TouchableOpacity')).toHaveLength(1)
            // console.log(wrapper)
            // expect(wrapper.length).toBe(1)
        })
    })
})