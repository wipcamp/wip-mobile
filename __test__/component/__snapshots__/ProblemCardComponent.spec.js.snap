// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottomWidth": 0.4,
      "borderColor": "#C8C8C8",
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flex": 0.2,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 0,
          "height": 0,
          "width": 0,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 0.8,
        },
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "flex-start",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": "Kanit",
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      Test Problem 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "Kanit",
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#858585",
          },
        ]
      }
    >
      Test Description 1
    </Text>
  </View>
</View>
`;
